uniffi_bindgen/src/bindings/gecko/templates/macros.cpp

```diff
@@ -0,0 +1,120 @@
+{# /* Functions that throw an error via `aRv` and return results by value still
+      need to return dummy values, even though they're discarded by the caller.
+      This macro returns a suitable "empty value". */ #}
+{%- macro bail(func) -%}
+{%- match func.return_type().and_then(self::default_return_value_cpp) -%}
+{%- when Some with (value) -%}
+return {{ value }};
+{%- else -%}
+return;
+{%- endmatch -%}
+{%- endmacro -%}
+
+{# /* Declares a return type for a function or method. Functions can return
+      values directly or via "out params"; this macro handles both cases. */ #}
+{%- macro decl_return_type(func) -%}
+  {%- match func.return_type() -%}
+    {%- when Some with (type_) -%}
+      {%- match ReturnPosition::for_type(type_) -%}
+        {%- when ReturnPosition::Return with (type_) -%} {{ type_|ret_type_cpp }}
+        {%- when ReturnPosition::OutParam with (_) -%} void
+      {%- endmatch -%}
+    {%- else -%} void
+  {%- endmatch -%}
+{%- endmacro -%}
+
+{# /* Declares a list of arguments for a WebIDL constructor. A constructor takes
```

Comment:
These lists look quite hideous in the generated code, but I haven’t figured out the right mix of `%}` and `-%}` to make Askama strip the right amount of whitespace so that argument lists look pretty. The fixed arguments, like `GlobalObject`, and inserting the right number of commas make this tricky. They are readable as they are, so maybe that’s good enough?